libraries/AP_Compass/examples/AP_Compass_test/AP_Compass_test.cpp

```diff
@@ -119,6 +132,9 @@ static void loop()
             hal.console->printf(" t=%u", (unsigned)read_time);
 
             hal.console->printf("\n");
+
+            // exit after 100-200 max attempts, that's enough to test it.
+            if ( max_attempts <= 0 ) { hal.scheduler->delay(100); exit(1); } 
```

Comment:
We do not exit in examples.

Reply:
really..?    I'm happy to change to a different construct for immediately giving-up if you point me at your prefered method ?